src/lib/replication/repl_dev/raft_state_machine.h

```diff
@@ -86,6 +86,8 @@ class StateMachineStore;
 #define RD_LOGE(...) RD_LOG(ERROR, ##__VA_ARGS__)
 #define RD_LOGC(...) RD_LOG(CRITICAL, ##__VA_ARGS__)
 
+static constexpr uint64_t snp_obj_id_type_mask = 0x8000000000000000;
```

Comment:
Let's add some comments to describe our design of obj_id encoding format here. Otherwise readers may be confused about the origin of this magic number.

And suggest using bit shifting instead of raw hex to represent these values much more clearly (same for `get_next_lsn`).

Reply:
Sure